test/atexit.jl

```diff
@@ -0,0 +1,152 @@
+using Test
+
+@testset "atexit.jl" begin
+    function _atexit_tests_gen_cmd_eval(expr::String)
+        cmd_eval = ```
+        $(Base.julia_cmd()) -e $(expr)
+        ```
+        return cmd_eval
+    end
+    function _atexit_tests_gen_cmd_script(temp_dir::String, expr::String)
+        script, io = mktemp(temp_dir)
+        println(io, expr)
+        close(io)
+        cmd_script = ```
+        $(Base.julia_cmd()) $(script)
+        ```
+        return cmd_script
+    end
+    atexit_temp_dir = mktempdir()
+    atexit(() -> rm(atexit_temp_dir; force = true, recursive = true))
+    @testset "these should exit with exit code 0" begin
+        julia_expr_list = Dict(
+            # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            """
+            atexit(() -> exit(0))
+            exit(22)
+            """ => 0,
+            # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            )
+        for julia_expr in keys(julia_expr_list)
+            cmd_eval = _atexit_tests_gen_cmd_eval(julia_expr)
+            cmd_script = _atexit_tests_gen_cmd_script(atexit_temp_dir, julia_expr)
+            expected_exit_code = julia_expr_list[julia_expr]
+            @test success(cmd_eval)
+            @test success(cmd_script)
+            p_eval = run(cmd_eval; wait = false)
+            p_script = run(cmd_script; wait = false)
+            wait(p_eval)
+            wait(p_script)
+            @test p_eval.exitcode == expected_exit_code
+            @test p_script.exitcode == expected_exit_code
+        end
+    end
+    @testset "these should exit with exit code 1" begin
+        julia_expr_list = Dict(
+            # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            """
+            atexit(() -> exit(1))
+            exit(22)
+            """ => 1,
+            # ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+            """
+            atexit(() -> println("No error"))
```

Comment:
Would you mind making a quick followup PR to make these silent during a normal test run?

```suggestion
            atexit(() -> "No error")
```

Reply:
Yeah!